smelli/unbinned.py

```diff
@@ -0,0 +1,78 @@
+from flavio.io import instanceio as iio
+import numpy as np
+import yaml
+
+class UnbinnedParameterLikelihood(iio.YAMLLoadable):
+    """An `UnbinnedParameterLikelihood` provides an unbinned likelihood function in terms of
+    parameters.
+    Methods:
+    - `log_likelihood`: The likelihood as a function of the parameters
+    Instances can be imported and exported from/to YAML using the `load`
+    and `dump` methods.
+    """
+    _input_schema_dict = {
+        'eft': str,
+        'basis': str,
+        'scale': float,
+        'names': list,
+        'samples': np.array,
+        'weights': np.array
+    }
+
+    _output_schema_dict = {
+        'eft': str,
+        'basis': str,
+        'scale': float,
+        'names': list,
+        'samples': np.array,
+        'weights': np.array
+    }
+
+    def __init__(self,
+                 eft,
+                 basis,
+                 scale,
+                 names,
+                 samples,
+                 weights=None):
+        """Initialize the instance.
+        Parameters:
+        - `eft`: a string identifying the WET as defined by wcxf
+        - `basis`: a string identifying the WET basis as defined by wcxf
+        - `scale`: a scale (in units of GeV) at which the WET coefficient samples are valid
+        - `names`: an ordered list of names identiying the Wilson coefficients in the WET basis
+        - `samples`: a numpy array of vectors of Wilson coefficients, with each row representing a sample
+        - `weights`: a numpy array of scalar weights on a logarithmic scale for each sample
+        """
+        self.eft     = eft
+        self.basis   = basis
+        self.scale   = scale
+        self.names   = names
+        self.samples = samples
+        self.weights = weights if weights is None else np.ones(len(samples))
```

Comment:
More Pythonic:
```python
self.weights = weights or np.ones(len(samples))
```

Reply:
Indeed. I had tested only with non-empty weights...

Reply:
@DavidMStraub your ore pythonic version does not work, since in the deserialization weights is provided as an ```np.array``` which has an implicit conversion to a boolean value (with warning). I will stick to the 'is not None'.